app/_src/production/cp-deployment/zoneproxy-auth.md

```diff
@@ -200,6 +200,102 @@ the new signing key.
+
+   ```sh
+   kumactl generate signing-key --format=pem > /tmp/key-private.pem
+   kumactl generate public-key --signing-key-path=/tmp/key-private.pem > /tmp/key-public.pem
+   ```
+
+   The result should be similar to this output
+   ```sh
+   cat /tmp/key-private.pem /tmp/key-public.pem 
+   -----BEGIN RSA PRIVATE KEY-----
+   MIIEpAIBAAKCAQEAsS61a79gC4mkr2Ltwi09ajakLyUR8YTkJWzZE805EtTkEn/r
+   ...
+   htKtzsYA7yGlt364IuDybrP+PlPMSK9cQAmWRRZIcBNsKOODkAgKFA==
+   -----END RSA PRIVATE KEY-----
+   -----BEGIN RSA PUBLIC KEY-----
+   MIIBCgKCAQEAsS61a79gC4mkr2Ltwi09ajakLyUR8YTkJWzZE805EtTkEn/rL2u/
+   ...
+   se7sx2Pt/NPbWFFTMGVFm3A1ueTUoorW+wIDAQAB
+   -----END RSA PUBLIC KEY----- 
+   ```
+
+2. Configure the control plane with public key
+
+   [Configure a control plane](/docs/{{ page.version }}/documentation/configuration) with the following settings
+   ```yaml
+   dpServer:
+     authn:
+       zoneProxy:
+         type: zoneToken
+         zoneToken:
+           enableIssuer: false # disable control plane token issuer that uses secrets
+           validator:
+             useSecrets: false # do not use signing key stored in secrets to validate the token
+             publicKeys:
+             - kid: "key-1"
+               key: |
+                 -----BEGIN RSA PUBLIC KEY-----
+                 MIIBCgKCAQEAsS61a79gC4mkr2Ltwi09ajakLyUR8YTkJWzZE805EtTkEn/rL2u/
+                 ...
+                 se7sx2Pt/NPbWFFTMGVFm3A1ueTUoorW+wIDAQAB
+                 -----END RSA PUBLIC KEY-----
+   ```
+
+3. Use the private key to issue tokens offline
+
+   The command is the same as with online signing, but with two additional arguments:
+   * `--kid` - ID of the key that should be used to validate the token. This should match `kid` specified in the control plane configuration.
+   * `--signing-key-path` - path to a PEM-encoded private key.
+
+   ```sh
+   kumactl generate zone-token \
+     --zone us-east \
+     --scope egress \
+     --valid-for 720h \
+     --signing-key-path /tmp/key-private.pem \
+     --kid key-1
+   ```
+
+   You can also use any external system that can issue JWT tokens using `RS256` signing method with the following claims:
+   * `Zone` (string) - the name of the zone
+   * `Scope` ([]string) - the list of scopes (`egress`, `ingress`)
+
+#### Migration
+
+You can use both offline and online issuing by keeping `dpServer.authn.zoneProxy.zoneToken.enableIssuer` to true.
+You can use both secrets and public key static config validators by keeping `dpServer.authn.zoneProxy.zoneToken.validator.useSecrets` to true.
+
+#### Management
+
+Token revocation works the same when using both online and offline issuing.
+
+Signing key rotation works similarly:
+* generate another pair of signing keys
+* configure a control plane with old and new public keys
+* regenerate tokens for all existing zone proxies with the new private key
+* remove the old public key from the configuration
+  {% endif_version %}
```

Comment:
nitpick: I don't think it matters but just in case let's not have the indent there:

```suggestion
{% endif_version %}
```